Stacked @card PR : (Card Reading Client)

valayDave opened the pull request.

valayDave committed:
Added Card Client.

valayDave committed:
Merge branch 'metaflow-cards-stacked-pr-base' into metaflow-cards-stacked-pr-client

valayDave committed:
Merge branch 'metaflow-cards-stacked-pr-base' into metaflow-cards-stacked-pr-client

valayDave committed:
Merge branch 'metaflow-cards-stacked-pr-base' into metaflow-cards-stacked-pr-client

valayDave committed:
Fixing bugs related to datastore root in client.

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+    def path(self):
+        return self._path
+
+    def browser(self):
```
change this to `view()` to make it consistent with the CLI

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+        self._temp_file = None
+
+    @property
+    def html(self):
```
change this to `get()` to make it consistent with the CLI. Also it is polite to keep properties as no-cost, side-effect-free operations, whereas this call actually makes a network request etc. - so let's drop `@property` on this one

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+        display(HTML(self.html))
+
+
+class CardIterator:
```
change this to `CardContainer`

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+        # return self._wrap_html()
+        return "\n".join(main_html)
+
+    def __next__(self):
```
let's drop `__next__`. If you want to iterate, you can call `iter(CardContainer)` or
`[card for card in container]`

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+        return self._get_card(self._current)
+
+
+def get_cards(task, type=None, follow_resumed=True):
```
as a matter of convenience, we could allow `task` to be a pathspec string optionally, so instead of `get_cards(Task('foo'))`, you can just do `get_cards('foo')`

valayDave replied:
I have added this too.

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+        url = "file://" + os.path.abspath(self._temp_file.name)
+        webbrowser.open(url)
+
+    def nb(self):
```
don't we need `_repr_html_` in this class too, so I can do
`get_cards('foo')[0]` and see the card in a cell?

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+    def _make_heading(self, type):
+        return "<h1>Displaying Card Of Type : %s</h1>" % type.title()
+
+    def _wrap_html(self, html):
```
is this used for anything?

valayDave replied:
Removed

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,191 @@
+        display_html(*main_html)
+
+    def _repr_html_(self):
+        from IPython.core.display import HTML
```
are these imports redundant?

valayDave replied:
Yes

valayDave replied:
Removed

valayDave committed:
Merge branch 'metaflow-cards-stacked-pr-base' into metaflow-cards-stacked-pr-client

valayDave committed:
Refactored client to based on ville's suggestions.
- `get_cards` supports pathspec as argument.

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,212 @@
+        display(HTML(self.get()))
+
+
+class CardContainer:
```
add a brief docstring here

valayDave replied:
Done.

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,212 @@
+_TYPE = type
+
+
+class Card:
```
add a brief docstring here

valayDave replied:
Done

tuulos commented on metaflow/plugins/cards/card_client.py:
```diff
@@ -0,0 +1,212 @@
+    @property
+    def path(self):
+        return self._path
+
```
add
```
def __str__(self):
       return "<Card at '%s'>" % self._path
```

valayDave replied:
Done

valayDave committed:
Merge branch 'metaflow-cards-stacked-pr-base' into metaflow-cards-stacked-pr-client

valayDave committed:
Added docstring to `Card`/`CardContainer`
- Remove `__next__` from `CardContainer`
- Remove `nb` from `Card`
- Remove `nb` from `CardContainer`
- Added `__str__` to `Card`

valayDave committed:
Merge branch 'metaflow-cards-stacked-pr-base' into metaflow-cards-stacked-pr-client

valayDave committed:
Fixed bug in client regarding card s3 root
- bug surface when no root is set (nor card or s3)

valayDave committed:
Merge branch 'metaflow-cards-stacked-pr-base' into metaflow-cards-stacked-pr-client
